src/main/java/uk/nhs/digital/nhsconnect/nhais/model/edifact/Interchange.java

```diff
@@ -1,14 +1,33 @@
 package uk.nhs.digital.nhsconnect.nhais.model.edifact;
 
 import lombok.Getter;
-import lombok.Setter;
 
 import java.util.List;
+import java.util.Optional;
 
-@Getter @Setter
+@Getter
 public class Interchange {
 
     // TODO: stub for the internal representation of an edifact interchange
-    private List<Segment> segments;
+    private final List<Segment> segments;
```

Comment:
Making List final doesn't make it immutable - constructor should copy or wrap it